webhook/webhook.go

```diff
@@ -0,0 +1,96 @@
+package webhook
+
+import (
+  "crypto/hmac"
+  "crypto/sha256"
+  "encoding/hex"
+  "net/http"
+  "fmt"
+  "errors"
+  "encoding/json"
+  "bytes"
+  "time"
+
+  log "github.com/gophish/gophish/logger"
+)
+
+const (
+  DefaultTimeoutSeconds = 10
+  MinHttpStatusErrorCode = 400
+  SignatureHeader = "X-Gophish-Signature"
+)
+
+
+//TODO
+
+type Sender interface {
+  Send(endPoint EndPoint, data interface{}) error
+}
+
+type defaultSender struct {
+  client *http.Client
+}
+
+var senderInstance = &defaultSender {
+  client: &http.Client{
+    Timeout: time.Second * DefaultTimeoutSeconds,
+    CheckRedirect: func(req *http.Request, via []*http.Request) error {
+      return http.ErrUseLastResponse
+    },
+  },
+}
+
+type EndPoint struct {
+  URL string
+  Secret string
+}
+
+func SendSingle(endPoint EndPoint, data interface{}) error {
```

Comment:
Personally, if we're calling `senderInstance.Send` here, I'd just rename this function to `Send` instead of `SendSingle`. I think it's cleaner.

Reply:
In that case we'll have 2 methods `Send` with almost the same signature, and interface with `Send` too. Do you think it'll be a good idea? 

I've called it `SendSingle` to distinguish it from `SendAll` and keep all that logical.

On the other hand, do we need to keep the method `Send` public  at all which now implements the interface `Sender`? If not, I'd rename it to `sendImpl`

Reply:
Correction: `func (ds defaultSender) Send(endPoint EndPoint, data interface{}) error {` doesn't even implement the interface `Sender`. We may want to remove the interface then. However, I said that the interface `Send` would be helpful when creating the tests for webhooks, right?

Reply:
It's quite ok to have both a package level `Send` and a default implementation level `Send`. If anything, I think it might help avoid ambiguity.

We can use [`net/http`](https://golang.org/pkg/net/http/) as our inspiration here- it has both a package level `Get`, `Post`, etc. that use the default client, but it also has individual client `Get`, `Post`, etc. This makes it clear to the user that you _can_ make your own client and use those methods, otherwise you can use the top-level methods which use the default client. We should follow the same pattern here.